detection_rules/devtools.py

```diff
@@ -918,7 +918,8 @@ def trim_version_lock(min_version: str, dry_run: bool):
             for outdated in outdated_vers:
                 popped = lock['previous'].pop(str(outdated))
                 if outdated == latest_version:
-                    lock['previous'][str(min_version)] = popped
+                    version_key = ".".join(map(str, min_version[:2]))
+                    lock['previous'][version_key] = popped
```

Comment:
`min_version` stored value is a tuple type, such as `(8, 3, 0)` and becomes `8.3.0` and becomes the key. This breaks schema as shown below where `definitions.SemVerMinorOnly` is defined below as well.

```sql
SemVerMinorOnly = NewType('SemVerFullStrict', str, validate=validate.Regexp(MINOR_SEMVER))
```

```sql
class VersionLockFileEntry(MarshmallowDataclassMixin, BaseEntry):
    """Schema for a rule entry in the version lock."""
    min_stack_version: Optional[definitions.SemVerMinorOnly]
    previous: Optional[Dict[definitions.SemVerMinorOnly, PreviousEntry]]
```

As a result, the first two elements in the tuple `(8, 2)` are spliced into a list and changed to strings, then joined by a decimal to produce `8.3` as a string type which passes validation.